game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/BattleStep.java

```diff
@@ -88,6 +90,7 @@
 
   static List<BattleStep> getAll(final BattleState battleState, final BattleActions battleActions) {
     return List.of(
+        new RemoveNonCombatantsBeforeAA(battleState, battleActions),
```

Comment:
It would be kinda nice to see 'removeNonCombatants' able to just do its thing and not have a variant that is pre-AA. Having the same battle step listed twice would be fine IMO (ie; this could be a `RemoveNonCombatants`), but the difference is that the object is able to compute at any given time, given a battle state, who the non-combatants are. This would make the object more useful and fully contain the logic of determining non-combatants (I think that would make for a stronger, more cohesive module. I also think the testing for it would be more interesting as well to determine that it computes non-combatants correctly across all cases).

Reply:
This is save compatibility issue.  I can't modify `RemoveNonCombatants` to give it additional logic because then an older save will still run it and do potentially strange things.  I also can't move `RemoveNonCombatents` for the same reason because an older save will have it in the old location.

Also, the `removeNonCombatants` logic inside of `MustFightBattle` is used in other places internally inside of `MustFightBattle` so moving it outside of `MustFightBattle` would either need to make it static method (which I think it can be) or duplicated (which I don't like the idea of).